### ISSUE TITLE
Ghost Name information

### ISSUE THREAD

**Issue by SpeedHighway** [azsry/phasmophobia_mechanics#7]
Part of the article states: "Ghosts can react to their name at any time, even when you are not holding down a VOIP key, with a 20 second delay in between reactions"

However, the code in question simply turns on a flag that the ghost's name has been said and turns that off after 20 seconds.  
`        if (SpeechRecognitionController.instance.hasSaidGhostsName)
        {
            this.ghostsNameReactionTimer -= Time.deltaTime;
            if (this.ghostsNameReactionTimer < 0f)
            {
                SpeechRecognitionController.instance.hasSaidGhostsName = false;
                this.ghostsNameReactionTimer = 20f;
            }
        }`

The section of code the uses this in the ResponseCheck and OnPhraseRecognized states:
`        if (!SpeechRecognitionController.instance.hasSaidGhostsName && !GameController.instance.isTutorial && Random.Range(0, 3) == 1 && PlayerPrefs.GetInt("isYoutuberVersion") == 0)
        {
            return;
        }`
Meaning, that if all of the following are true, for the player:
- Has not said the name in the past 20 seconds
- Is not in the tutorial
- Is not playing the "Youtuber" version
- If a randomly picked number, from set {0, 1, 2} is 1
Then the ResponseCheck fails.

What this says, to me, is that there is a roughly 66% chance of the ghost responding (provided all other checks, like same floor, correct room, correct ghost type, etc are met), which jumps to 100% if their name has been said in the last 20 seconds.  

What do you think?

**Comment by azsry** [comment]
> What this says, to me, is that there is a roughly 66% chance of the ghost responding (provided all other checks, like same floor, correct room, correct ghost type, etc are met), which jumps to 100% if their name has been said in the last 20 seconds.

I agree with you on this. I think I forgot to update that dot point, because I'd added the dot point below that which says

> Saying the ghost's name makes the ghost more likely to respond to the Spirit Box
and

> Ghosts are more likely to respond to the Spirit Box if you have said their name in isolation (i.e. anger them with a phrase) recently
in the Spirit Box section.

When I said the player doesn't need to be holding down the VOIP key, I was referring to the absence of any local push-to-talk checks like there are on all other phrase recognition systems.

Hope that helps!

**Comment by Mythra** [comment]
Hey! With the new update coming out, I'm just cleaning house of the repository, and closing old, stale issues. Seems this one was resolved so I'm going to go ahead, and close it, definitely feel free to re-open if you need!
